plutus-contract/src/Control/Monad/Freer/Log.hs

```diff
@@ -60,6 +65,26 @@ logWarn m = tell [LogMessage Warn m]
 logInfo :: Member Log effs => Text -> Eff effs ()
 logInfo m = tell [LogMessage Info m]
 
+-- | Write a log message before and after an action.
+surround :: Member Log effs => LogLevel -> Text -> Eff effs a -> Eff effs a
```

Comment:
And maybe we should consider using a more powerful logging library like katip or something. We could still proxy it though an effect, though.

Reply:
> We could still proxy it though an effect, though.

In `plutus-scb` we use [`monad-logger`](https://hackage.haskell.org/package/monad-logger-0.3.32) doing exactly that, but I agree about the structured log messages. It should be relatively painless since we can re-interpret the effects easily with `freer-simple`. I've been meaning to improve the logging situation for a while...